eta/core/datasets.py

```diff
@@ -1190,12 +1163,11 @@ def _extract_video_labels(self, start_frame, end_frame, labels):
         return segment
 
     def transform(self, src):
-        for entry in src.get_entries():
-            labels = entry.get_labels()
-            labels = self._extract_video_labels(entry.start_frame,
-                                                entry.end_frame, labels)
-            entry.set_labels(labels)
-        return src
+        for record in src.get_entries():
```

Comment:
BuilderDataset needs some method for allowing transformers to access it's data - and manipulate, replace, remove, and add to it